src/main/scala/viper/silver/plugin/standard/refute/RefuteErrors.scala

```diff
@@ -6,14 +6,17 @@
 
 package viper.silver.plugin.standard.refute
 
+import viper.silver.ast.FalseLit
 import viper.silver.verifier._
 
 sealed abstract class RefuteError extends ExtensionAbstractVerificationError
 sealed abstract class RefuteErrorReason extends ExtensionAbstractErrorReason
 
 case class RefuteFailed(override val offendingNode: Refute, override val reason: ErrorReason, override val cached: Boolean = false) extends RefuteError {
   override val id = "refute.failed"
-  override val text = "Refute holds in all cases or could not be reached (e.g. see Silicon `numberOfErrorsToReport`)."
+  override val text = if (offendingNode.exp.isInstanceOf[FalseLit])
+    "False could be proven here (which should never hold)." else
```

Comment:
What's the motivation for this change?

Reply:
The motivation is for the user to receive a more expressive error message. Since the `refute false` statements get added by the plugin when smoke detection is enabled, the user cannot see them in the Viper file. Hence, getting an error saying that some refute statement holds could be confusing, especially if the user does not know how the smoke detection plugin works internally, i.e. that it inserts `refute`s.
However, I guess the condition of the `if` could be more restrictive if we want the resulting error message to be applied only to `refute false` statements which are added by the plugin, i.e. by changing it to `offendingNode.exp.isInstanceOf[FalseLit] && offendingNode.info.getUniqueInfo[IdInfo].isDefined`. Should I implement this change?